addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -458,14 +458,32 @@ def test_mailing_unique_name(self):
         self.assertEqual(mailing_6.name, 'Second subject (Mass Mailing created on 2022-01-02)')
 
         mailing_0.subject = 'First subject'
-        self.assertEqual(mailing_0.name, 'First subject (Mass Mailing created on 2022-01-02) [4]',
+        self.assertEqual(mailing_0.name, 'First subject (Mass Mailing created on 2022-01-02)',
             msg='The name must have been re-generated')
 
-        mailing_0.name = 'Second subject (Mass Mailing created on 2022-01-02)'
+        mailing_0.name = 'Second subject (Mass Mailing created on 2022-01-02) [2]'
         self.assertEqual(mailing_0.name, 'Second subject (Mass Mailing created on 2022-01-02) [2]',
             msg='The name must be unique')
 
 
+    @freeze_time('2022-01-02')
+    @patch.object(Cursor, 'now', lambda *args, **kwargs: datetime(2022, 1, 2))
+    @users('user_marketing')
+    def test_mailing_name_update(self):
+        mailing_0 = self.env['mailing.mailing'].create({'subject': 'First subject'})
+        mailing_1, mailing_2 = self.env['mailing.mailing'].create([{'subject': 'First subject'}, {'subject': 'First subject'}])
+        mailing_3 = self.env['mailing.mailing'].create({'subject': 'First subject'})
+
+        self.assertEqual(mailing_0.name, 'First subject (Mass Mailing created on 2022-01-02)')
+        self.assertEqual(mailing_1.name, 'First subject (Mass Mailing created on 2022-01-02) [2]')
+        self.assertEqual(mailing_2.name, 'First subject (Mass Mailing created on 2022-01-02) [3]')
+        self.assertEqual(mailing_3.name, 'First subject (Mass Mailing created on 2022-01-02) [4]')
+        mailing_3.subject = 'Second subject'
+        self.assertEqual(mailing_3.name, 'Second subject (Mass Mailing created on 2022-01-02)')
+        mailing_3.subject = 'First subject'
+        self.assertEqual(mailing_3.name, 'First subject (Mass Mailing created on 2022-01-02) [4]')
+
```

Comment:
I'm not sure we're covering anything more than the test method above now

Maybe the update to the test above is sufficient? Seems to outline the goal of the change well enough.

Reply:
@reth-odoo test differst sliughtly: we change subject twice and second renaming is done in a way that matches already existing utms. But you are the boss here, so tomorrow think about it and i will change as you want

Reply:
I'd say it's fine to add a step in the test above for that then